### PR TITLE
feat: Flow components as children for ReactAdapterComponent

### LINKED ISSUE #19379: Support content elements in React adapter
Add support for adding Flow components as children to ReactAdapterComponent.

Implementation idea:

- add `useContent(name: string)` ReactAdapterElement hook, supports rendering a DOM placeholder in React
- add the corresponding `protected Element getContentElement(String name)` Java ReactAdapterComponent method, that creates and returns the named container element.

Prototype: https://github.com/vaadin/flow-hilla-hybrid-example/tree/proto/flow-react-children

The first implementation could use two DOM wrapper elements, each with style "display: contents;".

### PR DESCRIPTION
**Pull request by caalador**
Add support for having Flow Components
as children in a ReactAdapoterComponent

Fixes #19379

### CODE REVIEW

File: flow-react/src/main/java/com/vaadin/flow/component/react/ReactAdapterComponent.java [hunk 1]
```diff
@@ -49,6 +56,8 @@
  * @since 24.4
  */
 public abstract class ReactAdapterComponent extends Component {
+    private Map<String, Element> contentMap = new HashMap<>();
```
Comment (knoobie) [inline flow-react/src/main/java/com/vaadin/flow/component/react/ReactAdapterComponent.java:59]: Nit: Micro optimization: should this be null by default and only created if needed? I would think not so many react components going to have flow children

File: flow-client/src/main/java/com/vaadin/client/ReactUtils.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+     * @param element
+     *            react component element
+     * @param runnable
+     *            callback function runnable
+     */
+    public static native void addReadyCallback(Element element,
+            Runnable runnable)
+    /*-{
+            if(element.addReadyCallback){
+                element.addReadyCallback(
```
Comment (Legioth) [inline flow-client/src/main/java/com/vaadin/client/ReactUtils.java:45]: No `$entry` needed here?

File: flow-client/src/main/java/com/vaadin/client/ReactUtils.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+              function() {
```
Comment (Legioth) [inline flow-client/src/main/java/com/vaadin/client/ReactUtils.java:46]: Is the wrapping `function` necessary or could this be simplified as `element.addReadyCallback(runnable.@java.lang.Runnable::run(*).bind(runnable))``

File: flow-client/src/main/java/com/vaadin/client/ReactUtils.java (hunk 2 above)
Comment (caalador) [reply flow-client/src/main/java/com/vaadin/client/ReactUtils.java]: I guess it should. Added.

File: flow-client/src/main/java/com/vaadin/client/ReactUtils.java (hunk 3 above)
Comment (caalador) [reply flow-client/src/main/java/com/vaadin/client/ReactUtils.java]: :+1: removed extra function

File: flow-react/src/main/java/com/vaadin/flow/component/react/ReactAdapterComponent.java (hunk 1 above)
Comment (caalador) [reply flow-react/src/main/java/com/vaadin/flow/component/react/ReactAdapterComponent.java]: done.

**Review by tltv — DISMISSED** [review]
LGTM. Works as expected. Manually tested even ReactAdapterComponent inside a ReactAdapterComponent case works.

File: flow-react/src/main/resources/com/vaadin/flow/server/frontend/ReactAdapter.template [hunk 4]
```diff
@@ -118,6 +129,10 @@ export abstract class ReactAdapterElement extends HTMLElement {
         this.#maybeRenderRoot();
     }
 
+    public addReadyCallback(id: string, readyCallback: ReadyCallbackFunction) {
```
[the remark this answers is not in the document]
Comment (caalador) [reply flow-react/src/main/resources/com/vaadin/flow/server/frontend/ReactAdapter.template:142]: done

**Review by platosha — APPROVED** [review]

**Pull request merged**